server/routes/CheckRoleMiddleware.js

```diff
@@ -0,0 +1,10 @@
+function checkRole(roleArray) {
+    return function(req, res, next) {
+      if (req.user && roleArray.includes(req.user.role.name)) {
```

Comment:
I'm a bit confused about where req.user.role.name comes from. I understand req.user has the user data for the session, but where are we fetching it from the database and adding it to the request?

Reply:
good point, I forgot to mention this 

The fetching is done before this call is made. The fetched result from the database will be attached to the request as something like req.user and then this function checks if its Users' role is valid